### PR TITLE
api.php - auto-append extension

### PR DESCRIPTION
**Pull request by zertap**
Added paramater autoext to automatically append file extension to the name parameter supplied. (Useful for example in ShareX, if you want to have a custom filename, since you cannot supply the extension of the uploaded file in the uploader config.)

### CODE REVIEW

**Comment by zertap** [comment]
Fixed a stupid brainfart there. Hopefull works now. Well, this more more of a dirty fix than a real solution.

**Comment by nokonoko** [comment]
The problem with this is that you have to apply a dirty double dot checker (like on Pomf) in case of the file having 2 dots in extension e.g tar.gz, a better solution for this would just to add:

case 'upload-sharex':

and then do this since I (assume) ShareX is mostly for images which only has one dot in the extension, also in order to get only the extension and not fuck up filenames such as "image.2015.05.jpg" it's generally better to use $ext = pathinfo($filename, PATHINFO_EXTENSION);.

TL;DR I will not merge but will make a api section for sharex.

**Comment by nokonoko** [comment]
Err I missed the end(), nevermind I'll merge it for now.

**Comment by nokonoko** [comment]
Err my brain is leaking, so dumb, I'll whip something up, gimme a few.

**Comment by nokonoko** [comment]
After looking into ShareX some I found this:
https://github.com/ShareX/ShareX/wiki/Custom-Uploader-examples#uguuse

And after some fiddling i did:
After capure task->save image file as..

Then I took a screenshot, got a file save pop-up, named a .png as "test.jpg" and I got back the correct URL from Uguu with a custom name and file ext, so I don't really see the problem?

**Comment by zertap** [comment]
Try editing the config in destinations>custom uploaders>uguu, and add the "name" argument, inputting anything you want (for example,  %unix). It will return a not working link without an extension.

**Comment by alucard0134** [comment]
try identifing the mime and appending it that way? but i still fail to see an issue here as well tbh
